apps/server/src/modules/user/service/user.service.ts

```diff
@@ -40,7 +40,7 @@ export class UserService {
 
 	async createOrUpdate(user: UserDto): Promise<void> {
 		const userRoles: Role[] = await this.roleRepo.findByIds(user.roleIds);
-		const school: School = await this.schoolRepo.findById(user.schoolId);
+		const school: School = await this.schoolRepo.findById(user.schoolId as string);
```

Comment:
This is not good idea custom the type here. What happens if the user.schoolId is undefined? we need the other solution.

Please check it.

https://github.com/hpi-schul-cloud/schulcloud-server/pull/3589/commits/f1856525e1f469f291254c1c476cb980d90b023d
https://github.com/hpi-schul-cloud/schulcloud-server/pull/3589/commits/a9336263a1bf151acb381b161750d751ecf97af7

Reply:
now managed what is optional/mandatory to avoid these problems